### ISSUE TITLE
[Bug]: Broken text in flatpak GTK apps

### ISSUE THREAD

**Issue by KawaiiGirl2001** [flatpak/flatpak#5249]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.12.7

### What Linux distribution are you using?

Other (specify below)

### Linux distribution version

9.1

### What architecture are you using?

x86_64

### How to reproduce

1.Install any GTK app from flatpak 1.12.7 on RHEL 9.1. For example GIMP or "GNOME extension manager".
2. Run it.

### Expected Behavior

Normal display of text in applications is expected.

### Actual Behavior

All text in the app is broken.
![FlatpakBug](https://user-images.githubusercontent.com/111690251/211160987-c67fe7d6-471c-44f0-a496-18d753a407f1.png)

**Comment by justinjereza** [comment]
This bug affects me as well and I initially thought it was a problem specific to `org.gnome.Builder`. It wasn't until I tried to run `com.github.wwmm.easyeffects` and saw the same issue that I realized it might be a problem with `org.gnome.Platform` itself.

Issue #1097 mentions the same problem has happened in the past. Attempting to run something like `flatpak run --command=fc-cache org.gnome.Builder -fv` results in ``fc-cache: ../../src/fccache.c:808: FcCacheFini: Assertion `fcCacheChains[i] == NULL' failed.``

**Comment by alexpdp7** [comment]
I had the same issue. `flatpak run -vv` mentioned issues with the Cantarell fonts. I copied the Cantarell fonts from `/usr/share/fonts` to `~/.local/share/fonts` and that took care of the issue (using Gnome Calls).

**Comment by justinjereza** [comment]
Seems to be fixed in the 2023-04-26 build of `org.gnome.Platform/x86_64/44`.

**Comment by alexpdp7** [comment]
Yes, I rm'ed the Cantarell fonts from my home directory, and Gnome Calls now renders correctly.

(I had to restart my running GTK apps, because I guess they were using the fonts in the home directory, and they started having rendering issues.)
